src/server.js

```diff
@@ -289,10 +289,36 @@ function init(socket, client, generateToken) {
 			socket.emit("sign-out");
 		});
 
+		socket.on("push:register", (subscription) => {
+			if (!client.isRegistered() || !client.config.sessions[subscription.token]) {
+				return;
+			}
+
+			const registration = client.registerPushSubscription(client.config.sessions[subscription.token], subscription);
+
+			if (registration) {
+				client.manager.webPush.pushSingle(client, registration, {
+					timestamp: Date.now(),
+					title: "The Lounge",
+					body: "🚀 Push notifications have been enabled"
+				});
+			}
+		});
+
+		socket.on("push:unregister", (token) => {
+			if (!client.isRegistered()) {
+				return;
```

Comment:
I chose to implement a `isRegistered` function over simply doing `public` check to allow for mixed mode in the future (if we ever do it).